pkg/probe/http/http.go

```diff
@@ -28,16 +29,14 @@ import (
 )
 
 func New() HTTPProber {
-	return HTTPProber{&http.Client{}}
+	return HTTPProber{}
 }
 
-type HTTPProber struct {
-	client HTTPGetInterface
-}
+type HTTPProber struct{}
 
 // Probe returns a ProbeRunner capable of running an http check.
-func (pr *HTTPProber) Probe(host string, port int, path string) (probe.Status, error) {
-	return DoHTTPProbe(formatURL(host, port, path), pr.client)
+func (pr *HTTPProber) Probe(host string, port int, path string, timeout time.Duration) (probe.Status, error) {
+	return DoHTTPProbe(formatURL(host, port, path), &http.Client{Timeout: timeout})
```

Comment:
Is there any way this wouldn't have to create a client on every probe? They are somewhat expensive. Maybe, DoHTTPProbe could accept a timeout like DoTCPProbe, and then do a manual select After timeout in there.